### DISCUSSION TITLE
[Submarine Editor] Expanded tools for sprite/texture manipulation

### DISCUSSION THREAD

**Discussion by Xeavas** [FakeFishGames/Barotrauma#4805] [Ideas and suggestions]
Some suggestions on how to enhance the ability to manipulate texture sprites (decorative ones) with varying levels of complexity required to possibly achieve implementation.

### 1. Checkbox to disallow sprite texture repetition
Upon squishing something below its sprite boundaries (ie below 1 x/y scale), textures of said sprite would repeat, thus limiting the ability to shape some sprites to your liking (ie creating unique sprite blends and assemblies).

A checkbox is there for people who would also like to use the repeating textures to their liking, thus not limiting the utility of the x/y scales but rather expand it.

Below is an example(mockup) of what it would look like, you can see from the leftmost is a sprite with the y texture scale below 1, with the original sprite directly to the right of it. The 3rd sprite counting from the left is the same sprite with a y texture scale of 0.2, and the rightmost sprite is what said texture should look like if the checkbox is enabled, giving more utility for sub editors to decorate subs with greater ease and customizability

![examplemockup](https://user-images.githubusercontent.com/72674994/104912846-c5649780-59c7-11eb-90c8-3172df0607d4.jpg)

### 2. A more ui friendly/unified approach to sprite/texture scaling
Building off the first suggestion, making it function like what an image editor does (but to sprites/textures) would enhance the ability and ease to manipulate sprites & textures more, at the cost of development complexity and difficulty (when 3e849f2e5c was asked to comment about this).

It could potentially include sprite rotation into said unified tool too.

Example Below is what it would look like, most things are self-explanatory, with the x/y texture scale functioning differently if the "Disallow sprite repetition" checkbox is ticked or not, the rightmost sample would be an example of the checkbox being ticked.        

![mockup](https://user-images.githubusercontent.com/72674994/104913727-04471d00-59c9-11eb-8229-240c8f658f8d.jpg)

### 3. The application of previous two points on things which has/have collision boxes
I am doubtful of point 3. even being considered in the first place due to the difficulty & complexity of developing it, but it is here for food for thought anyway.

This would allow you to resize, shift, adjust textures independent of the collision box, potentially resizing and rotation of collision boxes itself, I didn't put much thought into this section due to mentioned dev intensive reasons.

**Comment by Psyduckdeluxe** [comment]
I just want this "could potentially include sprite rotation into said unified tool too."

But i feel rest of it is really more work than its worth :/

**Reply by Xeavas** [reply]
That's fair, I was primarily hoping point 1 would be considered, the rest is mostly a shot in the dark. 

-Could possibly make a new discussion just about rotation of decorative sprites.
